### PR TITLE
Add functionality to highlight the best match and take user directly to the match

### PR DESCRIPTION
**Pull request by caismith-ch**
- Add highlighting for the best match company
- Add js to take the user directly to best match
- Add fix for IOS company numbers showing as phone numbers

Resolves: GCI-230

### CODE REVIEW

File: src/controllers/search.controller.ts [hunk 1]
```diff
@@ -25,13 +25,28 @@ const route = async (req: Request, res: Response) => {
...
+              classes: "nearest"
+            },
+            {
+              text: result.items.company_number,
+            },
+            {
+              text: capitalisedStatus,
+            },
+          ];
+        }
```
Comment (nabeelamirch) [inline src/controllers/search.controller.ts]: The the text attributes are repeated in both returns. I would do something like this:
`let html = `<a href="${result.links.self}">${result.items.corporate_name}</a>`;
let classes = "";
if (result.items.corporate_name === topHit) {
    html =`<a id="centre-page" href="${result.links.self}">${result.items.corporate_name}</a>`,
    classes="nearest"
}

return [
    {
        html,
        classes
    },
    {
        text: result.items.company_number,
    },
    {
        text: capitalisedStatus,
    },
];`

Comment (caismith-ch) [reply src/controllers/search.controller.ts]: Thats a good shout, wasn't sure how to best do this but that definitely looks more concise and maintainable!

Comment (nabeelamirch) [reply src/controllers/search.controller.ts]: Cool, the change you just made, is the html supposed to different. In the previous commit the top hit had an id of "centre page". ALso I realised the code I suggested didn't quite paste properly;
```
let html = `<a href="${result.links.self}">${result.items.corporate_name}</a>`;
let classes = "";
if (result.items.corporate_name === topHit) {
    html =`<a id="centre-page" href="${result.links.self}">${result.items.corporate_name}</a>`,
    classes="nearest"
}

return [
    {
        html,
        classes
    },
    {
        text: result.items.company_number,
    },
    {
        text: capitalisedStatus,
    },
];
```

Comment (caismith-ch) [reply src/controllers/search.controller.ts]: Yeah sorry I noticed that I wasn't using the "centre-page" ID it was a leftover when I was testing it

Comment (nabeelamirch) [reply src/controllers/search.controller.ts]: Alright cool, looks fine to me

**Pull request merged**